src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -326,7 +388,97 @@ public void RequestLicKey(int dwReserved, [MarshalAs(UnmanagedType.BStr)] out st
                 [MarshalAs(UnmanagedType.BStr)] string bstrKey,
                 [MarshalAs(UnmanagedType.Interface)] out object ppvObject)
             {
-                throw new NotImplementedException();
+                Debug.Assert(pUnkReserved == null);
+                this.CreateInstanceInner(pUnkOuter, ref riid, bstrKey, isDesignTime: false, out ppvObject);
+            }
+
+            private void CreateInstanceInner(
+                object pUnkOuter,
+                ref Guid riid,
+                string key,
+                bool isDesignTime,
+                out object ppvObject)
+            {
+                BasicClassFactory.ValidateInterfaceRequest(this.classType, ref riid, pUnkOuter);
+
+                ppvObject = this.licenseProxy.AllocateAndValidateLicense(this.classType, key, isDesignTime);
+                if (pUnkOuter != null)
+                {
+                    ppvObject = BasicClassFactory.CreateAggregatedObject(pUnkOuter, ppvObject);
+                }
+            }
+        }
+
+        private class LicenseInteropProxy
+        {
+            private static readonly Type s_LicenseAttrType;
+            private readonly Type helperType;
+            private readonly MethodInfo getLicInfo;
+            private readonly MethodInfo requestLicKey;
+            private readonly MethodInfo allocateAndValidateLicense;
+            private readonly object instance;
+
+            static LicenseInteropProxy()
+            {
+                s_LicenseAttrType = Type.GetType("System.ComponentModel.LicenseProviderAttribute, System", throwOnError: true);
```

Comment:
`System` is a desktop compat shim. It would be better to use the actual assembly that the type is exposed in .NET Core - `System.ComponentModel.TypeConverter`.